src/Gaufrette/Stream/InMemoryBuffer.php

```diff
@@ -9,28 +9,25 @@
 
 class InMemoryBuffer implements Stream
 {
-    private $filesystem;
-    private $key;
-    private $mode;
-    private $content;
-    private $numBytes;
-    private $position;
-    private $synchronized;
+    private Filesystem $filesystem;
+    private string $key;
+    private StreamMode $mode;
+    private string $content;
+    private int $numBytes;
+    private int $position;
+    private bool $synchronized;
```

Comment:
I think only apply this typehint will create some troubles. For example, L191 will create failure instead of `return false` like currently.